plugin/storage/es/spanstore/reader.go

```diff
@@ -200,8 +200,10 @@ func timeRangeIndices(indexName, indexDateLayout string, startTime time.Time, en
 	firstIndex := indexWithDate(indexName, indexDateLayout, startTime)
 	currentIndex := indexWithDate(indexName, indexDateLayout, endTime)
 	for currentIndex != firstIndex {
-		indices = append(indices, currentIndex)
-		endTime = endTime.Add(-24 * time.Hour)
+		if len(indices) == 0 || indices[len(indices)-1] != currentIndex {
+			indices = append(indices, currentIndex)
+		}
+		endTime = endTime.Add(-1 * time.Hour)
```

Comment:
So this will do 24 iterations on every invocation, even though the default is daily rotation? Could we make this more efficient?

Reply:
ok let me take a look